### ISSUE TITLE
Doubts about the loss 

### ISSUE THREAD

**Issue by DabiaoMa** [CSTR-Edinburgh/merlin#147]
Hi, all

I appreciate this wonderful project and am still studying this model. While the behavior of the loss reduction really confused me. 

The input dimension of my case is about 410, the output  dimension of the acoustic features is 187. The initial validation loss is about 180+ as expected. While it seems like I can get pretty good test results even when the validation loss is only reduced to 174, which means that only a few features are close to the ground truth.  

Could anybody explain this to me?

**Comment by gillesdegottex** [comment]
Hi,

Keep in mind that the loss is the sum of all the feature dimensions.

Some dimensions might be pretty well predicted and their corresponding loss very low.
Some others, those badly correlated to the inputs, will have their corresponding loss very high (close to 1.0 if outputs are Gaussian-normalised).

Thus, a loss falling very little after training just tells us that most of the output dimensions are useless to the training. The predicted values of these dimensions will be zero (we minimize the MSE) and the denormalized predicted values will be equal to the mean measured in the training data during normalisation, which should not destroy the good predictions of any other values.

For example, the 2nd order approx. time derivative of the MGC is very likely to be complete garbage, and thus almost uncorrelated and unpredictable from the input labels.
But apparently the prediction of the lower cepstral coefs is enough for the resulting quality.

**Comment by DabiaoMa** [comment]
@gillesdegottex 
Thanks a lot, doubts cleared!
